crates/rust-analyzer/src/line_index.rs

```diff
@@ -18,6 +18,14 @@ pub(crate) struct LineIndex {
     pub(crate) encoding: OffsetEncoding,
 }
 
+impl LineIndex {
+    pub(crate) fn with_default_options(index: Arc<ide::LineIndex>) -> Self {
```

Comment:
We generally try to avoid defaults in rust-analyzer, and always let the caller to specify everything they need explicitly.

Reply:
I don't know what is difference of `ide::LineIndex` and `line_index::LineIndex`, so maybe it is even wrong. If it is ok for lsif, I just need to move this function to the `lsif.rs`?

Reply:
`ide::LineIndex` works for UTF-8 encoded files with Unix line endings as used internally by rust-analyzer. `line_index::LineIndex` supports both UTF-8 and UTF-16 encodings and Unix and Windows line endings. Lsp without the UTF-8 extension requires all offsets to be relative to an UTF-16 encoded version of the file. In addition it needs to use the actual line ending of the file. The split exists to be able to convert between the internal offsets and offsets lsp expects. The defaults you are giving here are likely incorrect.